Yafc.UI/ImGui/ScrollArea.cs

```diff
@@ -148,29 +148,43 @@ public float scrollX {
         protected abstract Vector2 MeasureContent(float width, ImGui gui);
 
         public bool KeyDown(SDL.SDL_Keysym key) {
-            switch (key.scancode) {
-                case SDL.SDL_Scancode.SDL_SCANCODE_UP:
+            bool ctrl = InputSystem.Instance.control;
+            bool shift = InputSystem.Instance.shift;
+            switch ((ctrl, shift, key.scancode)) {
+                case (false, false, SDL.SDL_Scancode.SDL_SCANCODE_UP):
                     scrollY -= 3;
                     return true;
-                case SDL.SDL_Scancode.SDL_SCANCODE_DOWN:
+                case (true, false, SDL.SDL_Scancode.SDL_SCANCODE_UP):
+                    scrollY = 0; // ctrl+up = home
+                    return true;
+                case (false, false, SDL.SDL_Scancode.SDL_SCANCODE_DOWN):
                     scrollY += 3;
                     return true;
-                case SDL.SDL_Scancode.SDL_SCANCODE_LEFT:
+                case (true, false, SDL.SDL_Scancode.SDL_SCANCODE_DOWN):
+                    scrollY = maxScroll.Y; // ctrl+down = end
+                    return true;
```

Comment:
The following passages are not a complaint against this PR, but a note in general. 
There are a lot of `return true` in this switch that are redundant because what we want is to return `true` if we have a corresponding command and `false` otherwise.

However, if we remove these returns, then the cases will "fall through", triggering all the cases below the one that was initially triggered, which will break stuff.

Do you guys have any ideas in this regard? There's [switch lambdas](https://medium.com/@aldyputu/java-switch-lambda-6da508e359b4) in Java 14, but I'm not sure if there's something similar in C#.

I found [pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression), but I'm not sure if it's the best solution.

Reply:
When the return true, the key press is mark as 'handled' and not send to its parent components. So basically the keypress is understood and processed.

Each different keypress is (obviously) handled different, so a 'fallthrough olution' wouldn't be possible as it would activate as if the other keypress would have been active as well.

In stead we could have
*  a `bool handled = false` and set it to true on each handled keypress, and after the switch check the value and return true/false. But I doubt this is much nicer, as it adds even more boilerplate...
* a 'map' of key/value pairs with 'keypress' (with modifiers, etc) as key and the action (lambda) as value. Check if the 'keypress' is in the map and be able to immediately see if the keypress is known or not. It reduces the boilet plate a little, but I doubt how readable it becomes...

Reply:
> a `bool handled = false` and set it to true on each handled keypress

How about pattern matching of `SDL_Scancode` and a `bool handled = true` that is set as `false` only by `default`, and then we check it only once before returning?

That allows for one-liner cases.

Reply:
Could be a nice solution yes

Reply:
> How about pattern matching of SDL_Scancode, and a bool handled = true that is set as false only by default, and then we check it only once before returning?

This would still require a `break;` statement in each case. Alternatively, we can switch to the expression `switch`, but that will make it a PITA if someone wants to add a multistatement case.

Reply:
> This would still require a `break;` statement in each case. Alternatively, we can `switch` to the expression switch, but that will make it a PITA if someone wants to add a multistatement case.

I'm not sure about breaks, but it turns out you're right about PITA. 
There is a problem with pattern matching for our case -- it is geared towards a sinlge command per switch. For instance, here is an example from a Microsoft [tutorial](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching):
```
public State PerformOperation(ReadOnlySpan<char> command) =>
   command switch
   {
       "SystemTest" => RunDiagnostics(),
       "Start" => StartSystem(),
       "Stop" => StopSystem(),
       "Reset" => ResetToReady(),
       _ => throw new ArgumentException("Invalid string value for command", nameof(command)),
   };
```

Overally, even if there is a way to compress the `switch` itself, it will require more code, which will not improve readability.